### ISSUE TITLE
Appium Compatibility / Accessibility Changes

### ISSUE THREAD

**Issue by henrymoulton** [storybookjs/react-native#153]
**Is your feature request related to a problem? Please describe.**

To enable Visual Testing using AWS Device Farm, we're looking at using Appium to click through our Storybook and take screenshots on every story. We would ideally choose Detox, however Device Farm doesn't support Detox due to Detox not supporting real devices.

If this solution works, it will enable automated Visual Testing using real devices and React Native, without requiring the Storybook Server to be running. 

At the moment Appium and React Native work together by making use of the `accessibilityLabel` rather than `testID` (see more [here](https://appiumpro.com/editions/76-testing-react-native-apps-with-appium) and [here](https://www.browserstack.com/guide/appium-react-native-for-automation) )

I can see that the OnDeviceUI of Storybook currently uses `testIDs` https://github.com/storybookjs/react-native/blob/master/app/react-native/src/preview/components/OnDeviceUI/navigation/button.tsx#L34

which enables it to test using Detox: https://github.com/storybookjs/react-native/blob/master/examples/native/e2e/story-list.spec.ts

**Describe the solution you'd like**

I would add `testID` and `accessibilityLabel` to various parts of the Storybook UI, including:

https://github.com/storybookjs/react-native/blob/master/app/react-native/src/preview/components/StoryListView/index.tsx#L180

https://github.com/storybookjs/react-native/blob/master/app/react-native/src/preview/components/OnDeviceUI/navigation/button.tsx#L34

**Describe alternatives you've considered**

I've tried selecting stories using `class chain` and `predicate string` selectors but have found them to be inaccurate. 
https://webdriver.io/docs/selectors/#ios-uiautomation

**Are you able to assist bring the feature to reality?**

Yes, happy to make a PR.

**Comment by henrymoulton** [comment]
Actually just seen

https://github.com/storybookjs/react-native/blob/master/app/react-native/src/preview/components/StoryListView/index.tsx#L60

for the `StoryListView`

**Comment by dannyhw** [comment]
@henrymoulton hey, does the id in the story list view solve it for you or do you think it would be good to have more acessibilityIDs? I would be happy to accept these changes if you would like to make a PR just let me know if you need any help or direction :)

**Comment by henrymoulton** [comment]
Believe a few more a11yIds would help :)

I'll make the PR soon, thanks for replying!

**Comment by tlomozik** [comment]
Hey @dannyhw and Storybook community! Since the new layout have came, I'm not able to do end-to-end tests 😕.  I was using Detox for screenshot generation (docs).
In versions <8 there were a11y IDs like `Storybook.ListView.SearchBar` or  `BottomMenu.Sidebar`, `BottomMenu.Canvas` and it all worked like a charm, but now, I was able to find only a11y ID for this: 

![Image](https://github.com/user-attachments/assets/6b08aa25-4a20-4973-bd52-77f5f9b2f088)

I checked codebase and I couldn't see any testID or accessibilityLabel that would allow me to open controls or search for a component. 

Would you be able to assist? I'm happy to make PR :)

Thanks for all great work that you've done so far for Storybook community.

**Comment by dannyhw** [comment]
I'm open to any prs for adding test ids

these components are mostly in packages/react-native-ui 

you can find it in `packages/react-native-ui/src/Layout.tsx` line 149

**Comment by dannyhw** [comment]
A note more generally about testing

I actually think that a better way to do e2e will be to send events to storybook via the websocket connection and turning off the storybook ui

I have some examples of this but just haven't but them together in a user friendly way

**Comment by tlomozik** [comment]
I considered this idea as well, but ultimately dropped it. I wasn't quite sure how I could utilize WebSockets in this scenario. If you have any resources or materials that could help me understand better, I'd greatly appreciate it. Additionally, I also do visual regression tests, which rely on Storybook UI, so test runner has to go through each story by clicking at sidebar => component => canvas etc. 

If you're saying I can send events like _open this story_ , _open another story_ via websockets that would be great!!!!

**Comment by dannyhw** [comment]
@tlomozik yes its built into react native storybook, you just need to provide the port to connect to and react native storybook will use that as the communication layer.

basically you can then just send events on that websocket server and it will go to all subscribed storybook instances

heres one example project I made with it:

https://github.com/dannyhw/rn-storybook-testing-poc/blob/main/e2e/starter.test.ts

Like I mentioned though its not really user friendly or well documented yet but its being worked on.

I see making testing options available as the next focus of rn storybook

**Comment by tlomozik** [comment]
As a follow-up for anyone interested. Websocket server for testing or screenshot making works great and it's so easy to setup. I have it integrated with Appium, doing visual regression tests. @dannyhw's testing poc repo helped a lot! Thank you again.

**Comment by dannyhw** [comment]
@tlomozik nice! Btw also adding an additional way to jump to stories with deep linking #691 see the pr for more details :)
